FWCore/Framework/src/TransitionProcessors.icc

```diff
@@ -242,18 +252,33 @@ private:
 
     iEP.readFile();
     iEP.respondToOpenInputFile();
+
+    iEP.inputProcessBlocks();
   }
 
   void gotoNewInputAndOutputFiles(EventProcessor& iEP) {
-    iEP.respondToCloseInputFile();
-    iEP.closeInputFile(false);
+    // If this is still true when we hit the destructor for
+    // the filesOpen_ object, then we will know an exception
+    // was thrown on one of the following 5 lines.
+    filesOpen_->closingSequenceAlreadyFailed_ = true;
 
+    iEP.respondToCloseInputFile();
+    bool cleaningUpAfterException = false;
+    iEP.closeInputFile(cleaningUpAfterException);
+    iEP.endProcessBlock(cleaningUpAfterException, filesOpen_->beginProcessBlockSucceeded_);
     iEP.closeOutputFiles();
 
+    // Initialize these for the next output file
+    filesOpen_->closingSequenceAlreadyFailed_ = false;
+    filesOpen_->beginProcessBlockSucceeded_ = false;
```

Comment:
Similarly, put the lines of code dealing with `beginProcessBlockSucceeded_` into its own block

Reply:
Done